src/Stripes.test.js

```diff
@@ -0,0 +1,123 @@
+import Stripes from './Stripes';
+
+describe('Stripes', () => {
+  describe('clone', () => {
+    it('returns a matching copy', () => {
+      const s = new Stripes();
+      const c = s.clone();
+      expect(s).toEqual(c);
+    });
+
+    it('returns a matching copy plus additional properties', () => {
+      const s = new Stripes();
+      const c = s.clone({ monkey: 'bagel' });
+
+      expect(c).toMatchObject(s);
+      expect(c).toHaveProperty('monkey', 'bagel');
+    });
+  });
+
+  describe('hasInterface', () => {
+    describe('returns undefined', () => {
+      it('before discovery', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger });
+        expect(s.hasInterface('monkey')).toBeUndefined()
+        expect(logger.log).toHaveBeenCalled();
+      });
+
+      it('before discovery.interfaces', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, discovery: {} });
+        expect(s.hasInterface('monkey')).toBeUndefined();
+        expect(logger.log).toHaveBeenCalled();
+      });
+
+      it('if the interface is absent', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, discovery: { interfaces: { 'funky': '3.0' }} });
+        expect(s.hasInterface('monkey')).toBeUndefined();
+        expect(logger.log).toHaveBeenCalledWith('interface', 'interface \'monkey\' is missing');
+      });
+    });
+
+    describe('returns true', () => {
+      it('when the interface is present and no version is provided', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, discovery: { interfaces: { 'monkey': '3.0' }} });
+        expect(s.hasInterface('monkey')).toBe(true);
+        expect(logger.log).toHaveBeenCalledWith('interface', 'interface \'monkey\' exists');
+      });
+
+      it('when the requested version is present', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, discovery: { interfaces: { 'monkey': '3.0' } } });
+        expect(s.hasInterface('monkey', '3.0')).toBe('3.0');
+        expect(logger.log).toHaveBeenCalled();
+      });
+    });
+
+    it('returns false when the requested version is absent', () => {
+      const logger = { log: jest.fn() };
+      const s = new Stripes({ logger, discovery: { interfaces: { 'monkey': '3.0' }} });
+      expect(s.hasInterface('monkey', '4.0')).toBe(0);
```

Comment:
Not really an issue for the present PR, but ...

It's weird that `hasInterface` returns 0 here instead of false.

Reply:
I think we should just document in the Stripes Developer Guide what the API actually is, and not start a fight with unknown existing code. (I'm 98% certain its return value is only ever checked for truthiness, but ... who knows? And in the Firefly world of course it's literally impossible to know.)